subprojects/core/src/main/java/org/gradle/api/internal/artifacts/JavaEcosystemSupport.java

```diff
@@ -270,4 +294,58 @@ public void execute(MultipleCandidatesDetails<Bundling> details) {
             }
         }
     }
+
+    @VisibleForTesting
+    public static class TargetPlatformCompatibilityRules implements AttributeCompatibilityRule<Integer>, ReusableAction {
+        @Override
+        public void execute(CompatibilityCheckDetails<Integer> details) {
```

Comment:
Can't we use `CompatibilityRuleChain#reverseOrdered()` instead?

Reply:
Indeed we can. And even more with disambiguation rules now that we don't have to deal with the "minimal version".